File: Data Structures & Algorithms/move-zeroes/submission-7.java
class Solution {
    public void moveZeroes(int[] nums) {
        int L = 0;
    

        // move L to the first zero position in nums 
        while (nums[L] != 0 && L < nums.length - 1) {
           L++;
        }

        int R = L+ 1;

        while (R < nums.length) {
            if (nums[R] != 0) {
                nums[L] = nums[R];
                nums[R] = 0; 
                L++;
            }
            R++;
        }
    }
}

File: Data Structures & Algorithms/majority-element/submission-7.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # lets try sorting first 

        nums.sort() 
        
        # this works because after we sort, the index at len(nums) // 2 will be the first occurance
        # of the majority element 
        return nums[len(nums) // 2]